packages/react/src/transactions/adapters/useCreateCommentController.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  PendingSigningRequestError,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import { CreateComment, CreateCommentRequest } from '@lens-protocol/domain/use-cases/publications';
+import { ProtocolCallUseCase } from '@lens-protocol/domain/use-cases/transactions';
+
+import { useSharedDependencies } from '../../shared';
+import { PromiseResultPresenter } from './PromiseResultPresenter';
+import { UploadHandler } from './UploadHandler';
+import { CommentCallGateway } from './publication-call-gateways/CommentCallGateway';
+
+export type UseCreateCommentControllerArgs = {
+  upload: UploadHandler;
+};
+
+export function useCreateCommentController({ upload }: UseCreateCommentControllerArgs) {
+  const {
+    apolloClient,
+    transactionFactory,
+    activeWallet,
+    transactionGateway,
+    protocolCallRelayer,
+    transactionQueue,
+  } = useSharedDependencies();
+
+  return async (request: CreateCommentRequest) => {
+    const gateway = new CommentCallGateway(apolloClient, transactionFactory, upload);
+
+    const presenter = new PromiseResultPresenter<
+      void,
+      PendingSigningRequestError | UserRejectedError | WalletConnectionError
+    >();
+
+    const signedCreateComment = new ProtocolCallUseCase<CreateCommentRequest>(
+      activeWallet,
+      transactionGateway,
+      gateway,
+      protocolCallRelayer,
+      transactionQueue,
+      presenter,
+    );
+
+    const createComment = new CreateComment(
+      signedCreateComment,
+      gateway,
+      transactionQueue,
+      presenter,
+    );
+
+    void createComment.execute(request);
```

Comment:
maybe we should await here?
When using the example web-wagmi app, the text input is cleared even before signing the first message in the wallet. If the user rejects the message, the comment is lost.

Reply:
I agree. When @desfero pointed out the issue with upload publication metadata errors and I looked into it, I ended up awaiting for this promise so to let possible exceptions during the upload bubble up to the `useCreatePost` hook. I think it should happen the same here and in `useCreateComment`.

Reply:
Right, was waiting for #61 to be merged to rebase and fix also catching the `uploadHandler` error.